### PR TITLE
WIP: Replacement ROI commands

### PR DESCRIPTION
**Pull request by DonLakeFlyer**
These changes are to:
* Support polygon structure scan (See #817)
* Replace existing ROI commands with ground station UI friendly variants (See #817)
* Deprecate replaced ROI commands

The current state of affairs is (from my poking through code, could be wrong):
* PX4
    * Supports both NAV_ROI and DO_SET_ROI
    * Supports MAV_ROI_LOCATION mode
    * I don't think it supports MAV_ROI_NONE to cancel an ROI
    * Does not support the remainder of the MAV_ROI_* modes
* ArduPilot
    * Supports DO_SET_ROI
    * Support MAV_ROI_LOCATION mode
    * Kind of supports MAV_ROI_NONE but in a strange way. By specifying 0's in lat/lon/lat for MAV_ROI_LOCATION
    * Does not support the remainder of the MAV_ROI_* modes

Given the above state of affairs, I have made the following changes:
* Added explicit MAV_CMD_DO_SET_ROI_LOCATION command which is a direct replacement for existing implementations of MAV_CMD_DO_SET_ROI+MAV_ROI_LOCATION.
* Added explicit MAV_CMD_DO_SET_ROI_NONE command which is a direct replacement for existing implementations of MAV_CMD_DO_SET_ROI+MAV_ROI_NONE.
* Add explicit new MAV_CMD_DO_SET_ROI_WPNEXT_OFFSET which implements the new support needed for polygonal structure scan.

I have specifially not included support for MAV_ROI_WPINDEX and MAV_ROI_TARGET since nobody has those implemented.

Related to closed pulls: #817, #818

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
Let's discuss these here and we can remove the WIP and merge when discussion complete. Especially interested in wording for _WPNEXT text. Not sure what I have is understandable.

**Comment by DonLakeFlyer** [comment]
FYI: This pull looks out of whack formatting wise and master-wise. I'll deal with that once discussion closes.

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -1029,6 +1029,36 @@
...
+		  <description>Sets the region of interest (ROI) to a location. This can then be used by the vehicles control system to control the vehicle attitude and the attitude of various sensors such as cameras.</description>
+		  <param index="1">Empty</param>
+		  <param index="2">Empty</param>
+		  <param index="3">Empty</param>
+		  <param index="4">Empty</param>
+		  <param index="5">Latitude</param>
+		  <param index="6">Longitude</param>
+		  <param index="7">Altitude</param>
+	  </entry>
+	  <entry value="195" name="MAV_CMD_DO_SET_ROI_WPNEXT_OFFSET">
```
Comment (LorenzMeier) [inline message_definitions/v1.0/common.xml]: These should not have the same value (195), see above.

**Comment by DonLakeFlyer** [comment]
Whoops.

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -1029,6 +1029,36 @@
...
+		  <param index="6">Longitude</param>
+		  <param index="7">Altitude</param>
+	  </entry>
+	  <entry value="196" name="MAV_CMD_DO_SET_ROI_WPNEXT_OFFSET">
+		  <description>Sets the region of interest (ROI) to be toward next waypoint, with optional pitch/roll/yaw offset. This can then be used by the vehicles control system to control the vehicle attitude and the attitude of various sensors such as cameras.</description>
+		  <param index="1">Empty</param>
+		  <param index="2">Empty</param>
+		  <param index="3">Empty</param>
+		  <param index="4">Empty</param>
+		  <param index="5">pitch offset from next waypoint</param>
```
Comment (amilcarlucas) [inline message_definitions/v1.0/common.xml]: Usually the order is roll, pitch , yaw

Comment (DonLakeFlyer) [reply message_definitions/v1.0/common.xml]: The order matches the DO_MOUNT_CONTROL order.

**Comment by DonLakeFlyer** [comment]
@amilcarlucas Is there some doc somewhere that explains what I need to do?

**Comment by DonLakeFlyer** [comment]
@rmackay9 @peterbarker I'm going to be pushing to close this out/merge this in soon. I need to get going on the new imlementation in QGC.

**Comment by LorenzMeier** [comment]
Rebased and cleaned up version: https://github.com/mavlink/mavlink/pull/823

**Comment by amilcarlucas** [comment]
@DonLakeFlyer I was asking you to reactivate your EXCELENT PR https://github.com/ArduPilot/pymavlink/pull/7 
That one is a truly great one, I would like to work with you on that.

**Comment by DonLakeFlyer** [comment]
I had completely forgotten about that one!

**Pull request closed without merging**